src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -167,12 +166,12 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 		if (!command.marker) {
 			throw new Error(`cannot add a decoration for a command ${JSON.stringify(command)} with no marker`);
 		}
-
 		const decoration = this._terminal.registerDecoration({ marker: command.marker });
 		if (!decoration) {
 			return undefined;
 		}
 		decoration.onRender(element => {
+			decoration.onDispose(() => this._decorations.delete(decoration.marker.id));
```

Comment:
before, the decorations weren't getting updated when:
clear -> marker dispose -> decoration dispose